dask/array/reductions.py

```diff
@@ -92,7 +93,8 @@ def partial_reduce(func, x, split_every, keepdims=False, dtype=None, name=None):
 
     >>> partial_reduce(np.min, x, {0: 1, 2: 3})    # doctest: +SKIP
     """
-    name = name or 'p_reduce-' + tokenize(func, x, split_every, keepdims, dtype)
+    name = name or funcname(func) + '-' + tokenize(func, x, split_every,
```

Comment:
We never passed in a `name` parameter, so a new name was computed each call in the loop. This invokes `tokenize` on the input array, and thus results in a different name. Just adding a paren around `(name or funcname(func)) + ...`. should actually solve this.